src/main.rs

```diff
@@ -450,110 +593,213 @@ fn get_setup(args: &[String]) -> Setup {
-                .opt_str(ALSA_MIXER_INDEX)
-                .map(|index| {
-                    index
-                        .parse::<u32>()
-                        .expect("Alsa mixer index is not a valid number")
+        });
+
+        #[cfg(not(feature = "alsa-backend"))]
+        let device = mixer_default_config.device;
+
+        #[cfg(feature = "alsa-backend")]
+        let index = matches
+            .opt_str(ALSA_MIXER_INDEX)
+            .map(|index| {
+                index.parse::<u32>().unwrap_or_else(|_| {
+                    error!(
+                        "Invalid `--{}` / `-{}`: {}",
+                        ALSA_MIXER_INDEX, ALSA_MIXER_INDEX_SHORT, index
+                    );
+                    println!("Default: {}", mixer_default_config.index);
+                    exit(1);
                 })
-                .unwrap_or(0),
-        };
+            })
+            .unwrap_or_else(|| mixer_default_config.index);
 
-        let control = match matches.opt_str("mixer-name") {
-            Some(name) => {
-                warn!("--mixer-name is deprecated and will be removed in a future release.");
-                warn!("Please use --alsa-mixer-control instead.");
-                name
-            }
-            None => matches
-                .opt_str(ALSA_MIXER_CONTROL)
-                .unwrap_or_else(|| MixerConfig::default().control),
-        };
+        #[cfg(not(feature = "alsa-backend"))]
+        let index = mixer_default_config.index;
+
+        #[cfg(feature = "alsa-backend")]
+        let control = matches
+            .opt_str(ALSA_MIXER_CONTROL)
+            .unwrap_or(mixer_default_config.control);
+
+        #[cfg(not(feature = "alsa-backend"))]
+        let control = mixer_default_config.control;
 
-        let mut volume_range = matches
+        let volume_range = matches
             .opt_str(VOLUME_RANGE)
-            .map(|range| range.parse::<f64>().unwrap())
+            .map(|range| {
+                let on_error = || {
+                    error!(
+                        "Invalid `--{}` / `-{}`: {}",
+                        VOLUME_RANGE, VOLUME_RANGE_SHORT, range
+                    );
+                    println!(
+                        "Valid `--{}` / `-{}` values: 0.0 - 100.0",
+                        VOLUME_RANGE, VOLUME_RANGE_SHORT
+                    );
+                    #[cfg(feature = "alsa-backend")]
+                    println!(
+                        "Default: softvol - {}, alsa - what the control supports",
+                        VolumeCtrl::DEFAULT_DB_RANGE
+                    );
+                    #[cfg(not(feature = "alsa-backend"))]
+                    println!("Default: {}", VolumeCtrl::DEFAULT_DB_RANGE);
+                };
+
+                let range = range.parse::<f64>().unwrap_or_else(|_| {
+                    on_error();
+                    exit(1);
+                });
+
+                if !(0.0..=100.0).contains(&range) {
```

Comment:
Final nit and again sorry if I could have mentioned this before. If you'd make these into constants like `MAX_VOLUME_RANGE` then you could re-use those in the `on_error` lines and not repeat yourself.

Reply:
Can do. I'll do that for all the ranges.

Reply:
Done.